frontend/src/utils/helpers/materials/extractMaterial.ts

```diff
@@ -0,0 +1,17 @@
+import { ContentElementType, IContentElement } from 'types/course';
+import { optimizeLink } from '../videoPlayer/videoLink';
+
+const extractMaterial = (contentElement: IContentElement): IContentElement['material'] => {
+  switch (contentElement.type) {
+    case ContentElementType.plain:
+      return contentElement.material;
+    case ContentElementType.video:
+      return optimizeLink(contentElement.material);
+    case ContentElementType.presentation:
+      return contentElement.material;
+    default:
+      throw new Error(`Unknown content element type: ${contentElement.type}.`);
+  }
+};
+
+export default extractMaterial;
```

Comment:
We be better to use something like this. With switch/case we mapping. Also it can be improved with '? :' i think
```suggestion
import { ContentElementType, IContentElement } from 'types/course';
import { optimizeLink } from '../videoPlayer/videoLink';

const extractMaterial = (contentElement: IContentElement): IContentElement['material'] => {
  const { type: contentType, material: contentMaterial } = contentElement;

  const returnedContentMaterial = {
    [ContentElementType.plain]: contentMaterial,
    [ContentElementType.presentation]: contentMaterial,
    [ContentElementType.video]: optimizeLink(contentMaterial),
  };

  const extractedContentMaterial = returnedContentMaterial[contentType];

  if (!extractedContentMaterial) {
    throw new Error(`Unknown content element type: ${contentType}.`);
  }

  return extractedContentMaterial;
};

export default extractMaterial;
```

Reply:
Do you think it will work? Or an error will be thrown when attempting to get unexisting array element?

Reply:
all should work fine. Error wil be trown, when we will have extractedContentMaterial = undefined.

Reply:
Just gimme a moment, I'll check it

Reply:
You're right! I will replace this logics with yours. Thanks! 🙇🏻